llama.cpp

```diff
@@ -5288,7 +5288,7 @@ struct llm_tokenizer_spm {
 
         // keep substituting the highest frequency pairs for as long as we can.
         while (!work_queue.empty()) {
-            auto bigram = work_queue.top();
+            const auto & bigram = work_queue.top();
             work_queue.pop();
```

Comment:
This looks like a bug - after `pop()` there is a dangling reference

Reply:
@ggerganov fix it https://github.com/ggerganov/llama.cpp/pull/3560/commits/574a6581ba7b5686a76f4c60bdf610c615c74d6d